src/index.ts

```diff
@@ -230,6 +258,246 @@ export class ParagraphAPI {
   ) {
     return this.api.getCoinHoldersByContract(contractAddress, params);
   }
+
+  /**
+   * Allows the user to buy a Pargraph coin
+   *
+   * @param
+   *  - coinId: ID of the coin to buy
+   *  - client: the Client object that is going to make the transaction
+   *  - account: the Account of the buyer
+   *  - amount: the amount of ETH in wei that is going to be swapped for the coin
+   * @returns
+   */
+  async buyCoin({
+    coinId,
+    client,
+    account,
+    amount,
+  }: {
+    coinId: string;
+    client: WalletClient;
+    account: Account;
+    amount: bigint;
+  }) {
+    const walletAddress = account.address;
+    const { args } = (await this.api.getBuyArgsById(coinId, {
+      walletAddress,
+      amount: amount.toString(),
+    })) as { args: [`0x${string}`, `0x${string}`[]] };
```

Comment:
do you know why you need to type this manually? why isn't the type specified in `this.api.getBuyArgsById`? ideally the openapi contract would have proper typing here